integration/integration_test.go

```diff
@@ -1163,89 +1148,71 @@ func testEscapeSequenceNoTrigger(t *testing.T, terminal *Terminal, sess <-chan e
-		// Person A types something into the terminal (including "exit")
-		personA.Type("\aecho hi\n\r\aexit\n\r\a")
-		sessionA <- cl.SSH(context.TODO(), []string{}, false)
-	}
+	go func() {
+		sessionA <- clientA.SSH(ctx, []string{}, false)
+	}()
 
-	// PersonB: wait for a session to become available, then join:
-	sessionB := make(chan error)
-	joinSession := func() {
-		sessionTimeoutCtx, sessionTimeoutCancel := context.WithTimeout(context.Background(), 10*time.Second)
-		defer sessionTimeoutCancel()
-		sessions, err := waitForSessionToBeEstablished(sessionTimeoutCtx, apidefaults.Namespace, site)
-		if err != nil {
-			sessionB <- trace.Wrap(err)
-			return
-		}
+	// wait for the session to become available
+	sess := waitForSessionToBeEstablished(ctx, t, site)
+	sid := sess.GetSessionID()
 
-		sessionID := string(sessions[0].ID)
-		cl, err := teleport.NewClient(helpers.ClientConfig{
-			Login:   username,
-			Cluster: helpers.Site,
-			Host:    Host,
-		})
-		if err != nil {
-			sessionB <- trace.Wrap(err)
-			return
-		}
+	// PersonB: join the session
+	clientB, err := teleport.NewClient(helpers.ClientConfig{
+		Login:   username,
+		Cluster: helpers.Site,
+		Host:    Host,
+	})
+	require.NoError(t, err)
 
-		timeoutCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-		defer cancel()
+	personB := NewTerminal(250)
+	clientB.Stdout = personB
+	clientB.Stdin = personB
 
-		ticker := time.NewTicker(100 * time.Millisecond)
-		defer ticker.Stop()
-		for {
-			select {
-			case <-timeoutCtx.Done():
-				sessionB <- timeoutCtx.Err()
-				return
+	sessionB := make(chan error)
+	go func() {
+		sessionB <- clientB.Join(ctx, types.SessionPeerMode, apidefaults.Namespace, session.ID(sid), personB)
+	}()
 
-			case <-ticker.C:
-				err := cl.Join(context.TODO(), types.SessionPeerMode, apidefaults.Namespace, session.ID(sessionID), personB)
-				if err == nil {
-					sessionB <- nil
-					return
-				}
+	// Person A types something into the terminal (including "exit")
+	personA.Type("\aecho hi\n\r\aexit\n\r\a")
+
+	// wait for the sessions to end without error
+	sessionEndedWithoutError := func(t *testing.T, errC chan error) func() bool {
+		return func() bool {
+			select {
+			case err := <-errC:
+				require.NoError(t, err)
+				return true
+			default:
+				return false
 			}
 		}
 	}
+	require.Eventually(t, sessionEndedWithoutError(t, sessionA), time.Second*10, time.Second)
+	require.Eventually(t, sessionEndedWithoutError(t, sessionB), time.Second*10, time.Second)
 
-	go openSession()
-	go joinSession()
-
-	// wait for the sessions to end
-	err := waitForError(sessionA, time.Second*10)
-	require.NoError(t, err)
-
-	err = waitForError(sessionB, time.Second*10)
-	require.NoError(t, err)
-
-	// make sure the output of B is mirrored in A
+	// make sure the output of A is mirrored in B
 	outputOfA := personA.Output(100)
 	outputOfB := personB.Output(100)
-	require.Contains(t, outputOfA, outputOfB)
+	require.Contains(t, outputOfB, outputOfA)
```

Comment:
This check doesn't actually do anything on master, as `outputOfB=""` since stdin/out isn't set above. Fixing that leads to the check failing, as the logic was flipped, since we are typing to A we expect the output in B. `outputOfB` has an extra symbol at the start `m╭─` which makes it so the two values aren't actually equal, which I assume is expected.